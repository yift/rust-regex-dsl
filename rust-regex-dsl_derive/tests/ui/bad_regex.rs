pub use rust_regex_dsl::regex_dsl;

fn main() {
    let regex = regex_dsl! {
        "|",
        regex {
            "[a-z+"
        },
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        regex {
            ""
        },
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        regex {
            123
        },
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        regex {
        },
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        regex,
        "|",
    };
    println!("{}", regex);
}