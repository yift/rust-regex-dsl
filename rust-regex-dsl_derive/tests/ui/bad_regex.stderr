error: regex parse error:
           [a-z+
           ^
       error: unclosed character class
 --> tests/ui/bad_regex.rs:7:13
  |
7 |             "[a-z+"
  |             ^^^^^^^

error: Empty regex is not supported
  --> tests/ui/bad_regex.rs:15:13
   |
15 |             ""
   |             ^^

error: expected string literal
  --> tests/ui/bad_regex.rs:23:13
   |
23 |             123
   |             ^^^

error: unexpected end of input, expected string literal
  --> tests/ui/bad_regex.rs:31:9
   |
31 |         },
   |         ^

error: Unknown word: regex
  --> tests/ui/bad_regex.rs:37:9
   |
37 |         regex,
   |         ^^^^^
